numpy/core/include/numpy/ufuncobject.h

```diff
@@ -173,7 +173,11 @@ typedef struct _tagPyUFuncObject {
          * but this was never implemented. (This is also why the above
          * selector is called the "legacy" selector.)
          */
-        vectorcallfunc vectorcall;
+        #ifndef Py_LIMITED_API
+            vectorcallfunc vectorcall;
```

Comment:
Maybe we should keep the version check as well as the Py_LIMITED_API one?  Something like:
```
#if !defined(Py_LIMITED_API) || Py_LIMITED_API >= 0x03080000 
```
because I would expect `vectorcallfunc` to be part of the limited API, just not in 3.6 of course.

Reply:
> Maybe we should keep the version check as well as the Py_LIMITED_API one? Something like:
> 
> ```
> #if !defined(Py_LIMITED_API) || Py_LIMITED_API >= 0x03080000 
> ```
> 
> because I would expect `vectorcallfunc` to be part of the limited API, just not in 3.6 of course.

According to the Python documentation and the latest cpython header files, `vectorcallfunc` is not part of the limited API at all --- in any version of Python. It's possible that it may be added to the limited API in a future version of Python, but I don't know how to guard against that.

> EDIT: As far as I understand, we need it so that an extension can be designed to be used on, e.g. Python 3.6 with an old version of NumPy, but _also_ on Python 3.8 with a new version (if compiled against that)?

Yes, but it's even more than that: third-party packages that use the limited API can build a single wheel targeting all Python versions for a given operating system (the `abi3` tag, see [PEP 425](https://www.python.org/dev/peps/pep-0425/)). I am the author of a package that does this (ligo.skymap - see wheels at https://pypi.org/project/ligo.skymap/#files). It's pretty slick!

> 
> EDIT2: We could probably also rename the `reserved2` to `vectorcall`. We are not lieing in the header, the user just doesn't know what the type is.

:+1: I'll make that change.

Reply:
Done.

Reply:
OK, thanks I had not expected it to be never part of the limited API.  LGTM then, @mattip may have an opinion about the test, we could potentially considering making it real test that runs in CI.